src/main/java/com/zhongan/devpilot/settings/DevPilotConfigForm.java

```diff
@@ -99,18 +141,53 @@ public DevPilotConfigForm() {
 
         modelComboBox = combo;
         comboBoxPanel = createOpenAIServiceSectionPanel(
-                DevPilotMessageBundle.get("devpilot.settings.service.modelTypeLabel"), modelComboBox);
+            DevPilotMessageBundle.get("devpilot.settings.service.modelTypeLabel"), modelComboBox);
 
         var instance = LanguageSettingsState.getInstance();
         index = instance.getLanguageIndex();
     }
 
+    public void refreshOllamaModels() {
+        ollamaModelComboBox.removeAllItems();
+        // 请求ollama接口,获取模型列表
+        String host = ollamaBaseHostField.getText();
+        if (null == host || "".equals(host)) {
+            JOptionPane.showMessageDialog(null, DevPilotMessageBundle.get("devpilot.settings.service.ollamaHostError"), DevPilotMessageBundle.get("devpilot.settings.service.dialog.error"), JOptionPane.ERROR_MESSAGE);
+            return;
+        }
+        if (host.endsWith("/")) {
+            host = host.substring(0, host.length() - 1);
+        }
+        try {
+            var request = new Request.Builder()
+                .get()
+                .url(host + "/api/tags")
+                .build();
+            Call call = OkhttpUtils.getClient().newCall(request);
+            okhttp3.Response response = call.execute();
+            if (response.isSuccessful()) {
+                var result = Objects.requireNonNull(response.body()).string();
+                JsonObject dataJson = JsonParser.parseString(result).getAsJsonObject();
+                JsonArray modelArray = dataJson.get("models").getAsJsonArray();
+                for (JsonElement modelEle : modelArray) {
+                    String modelName = modelEle.getAsJsonObject().get("name").getAsString();
+                    ollamaModelComboBox.addItem(modelName);
+                }
+            }
+            response.close();
+        } catch (Exception ex) {
+            // throw new RuntimeException(ex);
+            JOptionPane.showMessageDialog(null, DevPilotMessageBundle.get("devpilot.settings.service.ollamaHostError"), DevPilotMessageBundle.get("devpilot.settings.service.dialog.error"), JOptionPane.ERROR_MESSAGE);
```

Comment:
Maybe the api implement should in seperate utils or provider, it's strange that api call in form

Reply:
how about add a `listModels` method to LlmProvider? openai also has list modes api: `
https://api.openai.com/v1/models`, it's better to get models using an API instead of hardcoding it in OpenAIModelNameEnum. What is your opinion?

Reply:
It's all right, but it's better to have a default implement to prevent effect other implements

Reply:
please review again, thank you.